src/utils/__tests__/narrow-test.js

```diff
@@ -227,22 +228,41 @@ describe('isMessageInNarrow', () => {
 
     ['whole-stream narrow', streamNarrow(eg.stream.name), [
       ['matching stream message', true, eg.streamMessage()],
+      ['other-stream message', false, eg.streamMessage({ stream: otherStream })],
+      ['PM', false, eg.pmMessage()],
     ]],
     ['stream conversation', topicNarrow(eg.stream.name, 'cabbages'), [
       ['matching message', true, eg.streamMessage({ subject: 'cabbages' })],
+      ['message in same stream but other topic', false, eg.streamMessage({ subject: 'kings' })],
+      ['other-stream message', false, eg.streamMessage({ stream: otherStream })],
+      ['PM', false, eg.pmMessage()],
     ]],
 
     ['1:1 PM conversation, non-self', privateNarrow(eg.otherUser.email), [
-      ['matching PM', true, eg.pmMessage()],
```

Comment:
Under '1:1 PM conversation, non-self', should we also test for group-PM messages being excluded?